drivers/adc/adc_nrfx_saadc.c

```diff
@@ -725,6 +727,22 @@ static int adc_nrfx_read(const struct device *dev,
 {
 	int error;
 
+	m_data.correct_single_ended = true;
```

Comment:
should probably be done within `context_lock`, otherwise there is a risk of race condition. Same below